tests/stubs/switchstub.nim

```diff
@@ -26,12 +35,15 @@ method connect*(
  forceDial = false,
  reuseConnection = true,
  upgradeDir = Direction.Out) {.async.} =
-  if (self.connectStub != nil):
-    await self.connectStub()
+  if (self.connectStub != nil) and self.numStubExecutions < self.numStubToExecute:
```

Comment:
Why is this necessary? This logic could go in the `connectStub`

Reply:
Improved it.